### PR TITLE
Use "module" instead of "main" when exposing ESM.

### PR DESCRIPTION
**Pull request by fubhy**
In order for this package to work out of the box with webpack, we need to either name the index.js "index.mjs" (and not have a file extension, e.g. "esm/index") or use the "module" field in package.json. Otherwise it's not easily possible to load this through webpack without transpiling it.

### CODE REVIEW

**Comment by fubhy** [comment]
Ideally there would be different builds for "main" and "module" (e.g. see https://gist.github.com/jayphelps/51bafb4505558736fdba0aaf8bfe69d3)

File: package.json [hunk 1]
```diff
@@ -2,13 +2,16 @@
   "name": "rxjs-hooks",
   "version": "0.3.2",
   "description": "React hooks for RxJS",
-  "main": "esm/index.js",
+  "module": "dist/esm/index.js",
+  "main": "dist/cjs/index.js",
   "author": "LongYinan <lynweklm@gmail.com>",
   "license": "MIT",
   "typings": "esm/index.d.ts",
```
Comment (Brooooooklyn) [inline package.json]: this field should change to `dist/esm/index.d.ts` too

**Comment by fubhy** [comment]
Cool, thanks. Can you make a new release on npm? :-)

**Comment by zry656565** [comment]
@fubhy hi, 0.3.3 is published.

**Comment by fubhy** [comment]
Perfect, thanks a bunch!

**Pull request merged**